src/controller/auth.controller.ts

```diff
@@ -0,0 +1,137 @@
+        .escape(),
+    body('email', 'Family name must be specified.')
+        .trim()
+        .isLength({ min: 3 })
+        .withMessage(() => String(i18next.t('auth.user-invalid-email')))
+        .matches(/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/)
+        .withMessage(() => String(i18next.t('auth.user-invalid-email')))
+        .escape(),
+    body('password', 'Invalid password')
+        .matches(/^(?=.*[A-Z]).{8,}$/)
+        .withMessage(() => String(i18next.t('auth.user-invalid-password')))
+        .escape(),
+    body('repassword', 'Invalid password')
+        .optional({ checkFalsy: true })
+        .escape(),
+    asyncHandler(async (req: Request, res: Response) => {
+        try {
+            const errors = validationResult(req);
+            if (!errors.isEmpty()) {
+                res.render('auth/register', {
+                    errors: errors.array(),
+                    flash: req.t('auth.user-valid-name'),
+                });
+                return;
+            }
+            const { name, email, password, repassword } = req.body;
+            if (password !== repassword) {
+                req.flash('error', req.t('home.cfpassword-not-match'));
+                res.render('auth/register', {
+                    flash: req.flash() /* other variables */,
+                });
+                return;
+            }
+            const hashedPassword: string = await hashPassword(password);
+            const user: User = new User();
+            user.name = name;
+            user.email = email;
+            user.password = hashedPassword;
+            await UserRepository.save(user);
+            res.redirect('/auth/login');
+        } catch (error) {
+            req.flash('error', req.t('home.cant-create'));
+            res.redirect('/register');
+        }
+    }),
+];
+
+export const LoginGet = asyncHandler((req: Request, res: Response) => {
+    res.render('auth/login');
+});
+
+export const Login = [
+    body('email', 'Family name must be specified.')
+        .trim()
+        .isLength({ min: 3 })
+        .withMessage(() => String(i18next.t('form.user_invalid-email')))
+        .matches(/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/)
+        .withMessage(() => String(i18next.t('form.user_invalid-email')))
+        .escape(),
+    body('password', 'Invalid password')
+        .matches(/^(?=.*[A-Z]).{8,}$/)
+        .withMessage(() => String(i18next.t('form.user_invalid-password')))
+        .escape(),
+    asyncHandler(async (req: Request, res: Response) => {
+        try {
+            const errors = validationResult(req);
+            if (!errors.isEmpty()) {
+                res.render('auth/register', {
+                    errors: errors.array(),
+                });
+                return;
+            }
+            const { email, password } = req.body;
+            const userExist = await UserRepository.findOne({
+                where: { email: email },
+            });
```

Comment:
các logic liên quan đến xử lý trong DB em move ra module file rồi đặt trong folder `src/services` để tránh phình to action ở controller nhé